rust/worker/src/blockstore/arrow_blockfile/sparse_index.rs

```diff
@@ -0,0 +1,202 @@
+use crate::blockstore::types::BlockfileKey;
+use std::collections::{BTreeMap, HashMap};
+use std::fmt::Debug;
+use uuid::Uuid;
+
+/// A sentinel blockfilekey wrapper to represent the start blocks range
+#[derive(Clone, Debug)]
+enum SparseIndexDelimiter {
+    Start,
```

Comment:
Do we need `start`? We could just store the first key in the blockfile right? I'm fine either way, just checking my understandig